src/showingpreviously/cinemas/centre_for_the_moving_image.py

```diff
@@ -0,0 +1,134 @@
+    return r
+
+
+def get_days_to_fetch() -> [datetime.datetime]:
+    # Fetch films from the range of (today - DAYS_PREVIOUS) to today
+    base = datetime.date.today()
+    date_list = [base - datetime.timedelta(days=x) for x in range(DAYS_PREVIOUS)]
+    return date_list
+
+
+def get_showings_for_date(cinema: dict[str, any], fetch_date: datetime.datetime) -> [Showing]:
+    showings = []
+    fetch_date_str = fetch_date.strftime('%Y-%m-%d')
+    whats_on_url = f'{cinema["base_url"]}/whats-on/{fetch_date_str}'
+    req = get_response(whats_on_url)
+    soup = BeautifulSoup(req.text, features='html.parser')
+    showings_class = soup.find('div', class_='whats-on-list')
+    for showing_item in showings_class.find_all('div', class_='stub-details'):
+        showings += get_showing_item(cinema, fetch_date_str, showing_item)
+    return showings
+
+
+def get_showing_item(cinema: dict[str, any], showing_date_str: str, showing_item: BeautifulSoup) -> [Showing]:
+    showing_list = []
+    name = showing_item.find('span', class_='field--name-title').text
+    more_info_link = showing_item.find('a', class_='btn-more-info')['href']
+    release_year, attributes = get_film_page_details(cinema, more_info_link)
+    showing_times = showing_item.find('div', class_='screening-times-btns')
+    film = Film(name=name, year=release_year)
+    for showing_time in showing_times.find_all('a', class_='btn-times'):
+        showing_time_str = showing_time.text.strip().split('\n')[0].strip()
+        showing_datetime = datetime.datetime.strptime(f'{showing_date_str} {showing_time_str}', '%Y-%m-%d %H:%M')
+        showing_screen = Screen(name=showing_time.find('span', class_='screen').text.strip())
+        showing_type_attributes = get_showing_type_attributes(name, showing_date_str, showing_time)
+        showing = Showing(
+            film = film,
+            time = showing_datetime,
+            chain = CHAIN,
+            cinema = cinema['cinema'],
+            screen = showing_screen,
+            json_attributes = {**attributes, **showing_type_attributes}
+        )
+        showing_list.append(showing)
+    return showing_list
+
+
+def get_showing_type_attributes(film_name: str, showing_date_str: str, showing: BeautifulSoup) -> dict[str, any]:
+    showing_type_list = showing.find_all('span', class_='showing-type')
+    attributes = {}
+    for showing_type in showing_type_list:
+        showing_type_attribute = [attr for attr in showing_type['class'] if attr != 'showing-type']
+        if len(showing_type_attribute) != 1:
+            raise CinemaArchiverException('Expected one showing type attribute for {film_name} on {showing_date_str}: {str(showing_type_attribute)}')
+        showing_type_attribute = showing_type_attribute[0]
+        if showing_type_attribute not in KNOWN_SHOWING_TYPE_ATTRIBUTES:
+            raise CinemaArchiverException('Unknown showing type attribute for {film_name} on {showing_date_str}: {showing_type_attribute}')
```

Comment:
Do we want to raise an exception here? Or log it (somehow) and continue processing?

Reply:
We don't have a proper logging framework (yet) and I want to ensure that any new attributes are caught as they are introduced. Raising an exception seems to be the best way to do this, at least for now.